sparse/core.py

```diff
@@ -641,11 +650,13 @@ def elemwise(self, func, *args, **kwargs):
                    sorted=self.sorted)
 
     def elemwise_binary(self, func, other, *args, **kwargs):
-        assert isinstance(other, COO)
+        if not isinstance(other, COO):
+            return self.elemwise(func, other, *args, **kwargs)
```

Comment:
Thanks for this change.  I think that this is nicer, especially given that `elemwise_binary` is public API.  People will probably try to use it and your change here will, I suspect, include fewer surprises for them.

Reply:
We might at some point reduce this down to a single public `elemwise` function that handles both cases and dispatches to `_elemwise_unary` and `_elemwise_binary` based on inputs.  (not necessarily now though if you'd like to wrap this up quickly).

Reply:
No, properly > quickly. It saves headache down the line.

The problem I see with that is, `elemwise` already exists and does what you are describing as `_elemwise_unary`. I have no problem dispatching between the two, but my question is:

- Is it worthwhile changing the API? Changing what `elemwise` is and hiding `elemwise_binary`? Do we have someone for whom it'll break?
- I was also considering supporting `scipy.sparse.spmatrix`. It's a small change, and well worth it, I expect.

Reply:
It will also make implementing `__array_ufunc__` much easier.

Reply:
The new `elemwise` would, I think, cover the functionality of the old `elemwise`.  It would now *also* work if the args weren't scalars, but were other things as well.  

In general though this project is young enough and obscure enough that I think API breaks are fine.  Anyone using this project is an early adopter.

I agree that supporting `scipy.sparse.spmatrix` objects would be valuable.